diff -u -r ../Python-3.6.0/Modules/pwdmodule.c ./Modules/pwdmodule.c
--- ../Python-3.6.0/Modules/pwdmodule.c	2016-12-23 02:21:21.000000000 +0000
+++ ./Modules/pwdmodule.c	2016-12-27 04:37:26.157177253 +0000
@@ -82,7 +82,7 @@
 #endif
     PyStructSequence_SET_ITEM(v, setIndex++, _PyLong_FromUid(p->pw_uid));
     PyStructSequence_SET_ITEM(v, setIndex++, _PyLong_FromGid(p->pw_gid));
-#if defined(HAVE_STRUCT_PASSWD_PW_GECOS)
+#if defined(HAVE_STRUCT_PASSWD_PW_GECOS) && !defined(__ANDROID__)
     SETS(setIndex++, p->pw_gecos);
 #else
     SETS(setIndex++, "");
