--- ./configure	2017-03-26 21:07:51.000000000 +0530
+++ ../configure	2017-06-07 09:56:17.302265798 +0530
@@ -16677,30 +16677,8 @@
     ac_fn_c_check_header_mongrel "$LINENO" "gnutls/gnutls.h" "ac_cv_header_gnutls_gnutls_h" "$ac_includes_default"
 if test "x$ac_cv_header_gnutls_gnutls_h" = xyes; then :
 
-      if test "$cross_compiling" = yes; then :
-  { { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
-$as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
-as_fn_error $? "cannot run test program while cross compiling
-See \`config.log' for more details" "$LINENO" 5; }
-else
-  cat confdefs.h - <<_ACEOF >conftest.$ac_ext
-/* end confdefs.h.  */
-
-
-          #include <gnutls/gnutls.h>
-          #include <stdlib.h>
-int
-main ()
-{
-
-          exit(!(GNUTLS_VERSION_NUMBER >= 0x030308));
-  ;
-  return 0;
-}
-_ACEOF
-if ac_fn_c_try_run "$LINENO"; then :
-   { $as_echo "$as_me:${as_lineno-$LINENO}: checking for gnutls_init in -lgnutls" >&5
-$as_echo_n "checking for gnutls_init in -lgnutls... " >&6; }
+$as_echo "$as_me:${as_lineno-$LINENO}: checking for gnutls_init in -lgnutls" >&5
+$as_echo_n "checking for gnutls_init in -lgnutls... " >&6;
 if ${ac_cv_lib_gnutls_gnutls_init+:} false; then :
   $as_echo_n "(cached) " >&6
 else
--- ./configure	2017-06-07 09:58:04.428835683 +0530
+++ ../configure	2017-06-07 09:58:59.922117840 +0530
@@ -16741,8 +16741,6 @@
 $as_echo "no - GnuTLS support disabled" >&6; }
 fi
 
-  fi
-fi
 
 
   # Check whether --enable-libgeoip was given.
