diff --git a/util/meson.build b/util/meson.build
index d9a22a2..a269252 100644
--- a/util/meson.build
+++ b/util/meson.build
@@ -18,10 +18,7 @@ executable('mount.fuse3', ['mount.fuse.c'],
            install: true,
            install_dir: get_option('sbindir'))
 
-udev = dependency('udev')
-udevrulesdir = join_paths(udev.get_pkgconfig_variable('udevdir'), 'rules.d')
+udevrulesdir = '/etc/udev/rules.d'
 
-meson.add_install_script('install_helper.sh', get_option('sysconfdir'),
-                         get_option('bindir'), udevrulesdir)
 
 
