--- ./src/debug/createdump/dumpwriter.h	2018-05-14 23:07:38.000000000 +0530
+++ ./src/debug/createdump/dumpwriter.h	2018-06-10 10:54:11.696875252 +0530
@@ -1,7 +1,35 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
-
+struct prstatus_t {
+  struct elf_siginfo pr_info;
+  short pr_cursig;
+  unsigned long pr_sigpend;
+  unsigned long pr_sighold;
+  pid_t pr_pid;
+  pid_t pr_ppid;
+  pid_t pr_pgrp;
+  pid_t pr_sid;
+  struct timeval pr_utime;
+  struct timeval pr_stime;
+  struct timeval pr_cutime;
+  struct timeval pr_cstime;
+  elf_gregset_t pr_reg;
+  int pr_fpvalid;
+};
+#define ELF_PRARGSZ (80)
+struct prpsinfo_t {
+  char pr_state;
+  char pr_sname;
+  char pr_zomb;
+  char pr_nice;
+  unsigned long pr_flag;
+  __kernel_uid_t pr_uid;
+  __kernel_gid_t pr_gid;
+  pid_t pr_pid, pr_ppid, pr_pgrp, pr_sid;
+  char pr_fname[16];
+  char pr_psargs[ELF_PRARGSZ];
+};
 #ifdef BIT64
 #define ELF_CLASS ELFCLASS64
 #else
