--- ./src/Native/Unix/System.Native/CMakeLists.txt	2018-05-11 02:32:05.000000000 +0530
+++ ../CMakeLists.txt	2018-06-17 14:59:53.475944922 +0530
@@ -23,7 +23,7 @@
     pal_sysctl.cpp
 )
 
-if (CMAKE_SYSTEM_NAME STREQUAL Linux)
+if (CMAKE_SYSTEM_NAME STREQUAL Linux OR CMAKE_SYSTEM_NAME STREQUAL Android)
     set(NATIVE_SOURCES ${NATIVE_SOURCES} pal_networkchange.cpp)
 
     if (!HAVE_LINUX_RTNETLINK_H)
